test/extended/prometheus/prometheus.go

```diff
@@ -614,6 +597,32 @@ var _ = g.Describe("[sig-instrumentation] Prometheus [apigroup:image.openshift.i
 			}
 		})
 
+		g.It("should only discover targets via endpointslice", func() {
+			targets, err := fetchPrometheusTargets(context.Background(), prometheusURL, bearerToken)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			jobsUsingEndpoints := sets.NewString() // Use a set to avoid duplicate job names
+			for _, target := range targets.Data.ActiveTargets {
+				if _, ok := target.DiscoveredLabels["__meta_kubernetes_endpoints_name"]; ok {
+					if jobName, hasJob := target.Labels["job"]; hasJob {
+						jobsUsingEndpoints.Insert(jobName)
```

Comment:
I think that `namespace/job` to identify the job is more appropriate and will be more helpful